src/BuildScriptGenerator/Helpers/ParserHelper.cs

```diff
@@ -0,0 +1,58 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    using System.IO;
+    using Nett;
+    using Newtonsoft.Json.Linq;
+    using YamlDotNet.RepresentationModel;
+
+    /// <summary>
+    /// Helper class to parse various files into a native type.
+    /// </summary>
+    internal static class ParserHelper
+    {
+        /// <summary>
+        /// Parse a .toml file into a TomlTable from the Nett library.
+        /// See https://github.com/paiden/Nett for more information.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <param name="filePath">The path to the .toml file.</param>
+        /// <returns>A TomlTable object containing information about the .toml file.</returns>
+        public static TomlTable ParseTomlFile(ISourceRepo sourceRepo, string filePath)
+        {
+            var tomlContent = sourceRepo.ReadFile(filePath);
+            return Toml.ReadString(tomlContent);
+        }
+
+        /// <summary>
+        /// Parse a .yaml file into a YamlMappingNode from the YamlDotNet library.
+        /// See https://github.com/aaubry/YamlDotNet for more information.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <param name="filePath">The path to the .yaml file.</param>
+        /// <returns>A YamlMappingNode object containing information about the .yaml file.</returns>
+        public static YamlMappingNode ParseYamlFile(ISourceRepo sourceRepo, string filePath)
+        {
+            var yamlContent = sourceRepo.ReadFile(filePath);
+            var yamlStream = new YamlStream();
```

Comment:
Check if you need to dispose this stream

Reply:
This type doesn't have a `.Dispose()` or `.Close()` call (I don't know why this type is called a stream, it inherits from an enumerable)